src/main/java/org/openelisglobal/program/controller/ProgramController.java

```diff
@@ -39,6 +73,23 @@ public class ProgramController extends BaseRestController {
     @Autowired
     private TestSectionService testSectionService;
 
+    private static final List<String> EXCLUDED_PROGRAM_NAMES = List.of("Cytology", "Immunohistochemistry",
+            "Histopathology", "Pathology");
+
+    /**
+     * Returns all general programmes (excluding Cytology, Immunohistochemistry,
+     * Pathology) and their order entry data, but not the questionnaire.
+     */
+    @GetMapping(value = "/programs/general", produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseBody
+    public List<Program> getGeneralPrograms() {
```

Comment:
The current approach fetches all programs with programService.getAll() and filters them in memory using a Java stream. If the number of programs is large, this could be inefficient. I would add a filtering mechanism at the database or service layer, so only the relevant programs are returned. For example, create a method like `programService.getGeneralPrograms(EXCLUDED_PROGRAM_NAMES)` that performs exclusion in the query.